analyses/cnv-frequencies/01-cnv-frequencies.py

```diff
@@ -0,0 +1,249 @@
+     p.add_argument('ENSG_RMTL', type=str, default=None,
+                    help="OPenPedCan Ensembl to RMTL mapping file (ensg-hugo-rmtl-v*-mapping.tsv)\n\n")
+     p.add_argument('-v', '--version', action='version',
+                   version=f"classify_reads.py version {__version__} ({__date__})",
+                   help="Print the current 01-snv-frequencies.py version and exit\n\n")
+     return p.parse_args()
+
+
+def merge_histology_and_cnv_data(histology_file, cnv_consensus_file):
+	# load histology file
+	histology_df = pd.read_csv(histology_file, sep="\t", na_filter=False, dtype=str)
+	histology_df.head()
+
+	# subset histology dataframe for relevant columns
+	histology_df = histology_df[["Kids_First_Biospecimen_ID","Kids_First_Participant_ID", 
+	                                    "cohort", "cancer_group", "sample_type"]]
+
+	# load CNV consensus file
+	cnv_df = pd.read_csv(cnv_consensus_file, sep="\t")
+	#cnv_df.head()
+
+	# merge subset of histology dataframe to CNV dataframe keeping only sample
+	# present in the CNV table (left outer join)
+	merged_df = pd.merge(cnv_df, histology_df, how="left", left_on="biospecimen_id", 
+	                     right_on="Kids_First_Biospecimen_ID")
+
+	# check if non tumor sample are present in the merged dataframe 
+	if  not np.array_equal(np.array(["Tumor"]), merged_df.sample_type.unique()):
+	    raise Exception(f"Merged hsitology-CNV dataframe contains non tumor samples")
+
+	# check and drop unknown cancer types (cancer_group == NA)
+	row_indices = merged_df[(merged_df["cancer_group"] == "NA")].index
+	merged_df.drop(row_indices, inplace=True)
+	    
+	# select and reorder relevant columns 
+	all_tumors_df = merged_df[["gene_symbol", "ensembl", "status", "copy_number", "ploidy", "cohort",
+	                  "cancer_group", "Kids_First_Biospecimen_ID", "Kids_First_Participant_ID"]].reset_index()
+	return(all_tumors_df)
+
+
+def get_cancer_groups_and_cohorts(all_tumors_df):
+	# group samples by cancer_group and cohort
+	cancer_group_cohort_df = all_tumors_df.groupby(["cancer_group", "cohort"])["Kids_First_Biospecimen_ID"].nunique().reset_index()
+	cancer_group_cohort_df.columns = ["cancer_group", "cohort", "num_samples"]
+
+	# group samples by cancer_group only 
+	def func(x):
+	    d = {}
+	    cohort_list = x["cohort"].unique()
+	    if len(cohort_list) > 1:
+	        d["cohort"] = "all_cohorts"
+	    else:
+	        d["cohort"] = cohort_list[0]                  
+	    d["num_samples"] = x["Kids_First_Biospecimen_ID"].nunique()
+	    return pd.Series(d, index=["cohort", "num_samples"])   
+	cancer_group_df = all_tumors_df.groupby(["cancer_group"]).apply(func).reset_index()
+
+	# concat cancer_group_cohort_df and cancer_group_df and rremove duplicates
+	cancer_group_cohort_df = pd.concat([cancer_group_df, cancer_group_cohort_df], sort=False, ignore_index=True)
+	cancer_group_cohort_df.drop_duplicates(inplace=True, ignore_index=True)
```

Comment:
The `ignore_index=True` argument is not supported by the pandas version 0.25.3 on the docker image.

```suggestion
	cancer_group_cohort_df.drop_duplicates(inplace=True)
```

You may need to add additional code to implement the `ignore_index=True` effect, like `.reset_index(drop=True)`.

Reply:
No need to implement `ignore_index=True`,  In pandas 0.25.3 indexes are ignored by default. 
https://pandas.pydata.org/pandas-docs/version/0.25.3/reference/api/pandas.DataFrame.drop_duplicates.html?highlight=drop_duplicate
``` 
DataFrame.drop_duplicates(self, subset=None, keep='first', inplace=False)[source]
Return DataFrame with duplicate rows removed, optionally only considering certain columns. Indexes, including time indexes are ignored.
```